numcodecs/lz4.pyx

```diff
@@ -97,17 +101,15 @@ def compress(source, int acceleration=DEFAULT_ACCELERATION):
                                                 acceleration)
 
     finally:
-
-        # release buffers
-        source_buffer.release()
+        pass
 
     # check compression was successful
     if compressed_size <= 0:
         raise RuntimeError('LZ4 compression error: %s' % compressed_size)
 
     # resize after compression
     compressed_size += sizeof(uint32_t)
-    dest = dest[:compressed_size]
+    PyBytes_RESIZE(dest, compressed_size)
```

Comment:
Thus when we need to truncate a `bytes` object, we can avoid `[:<n>]`, which copies the `bytes` object

Instead we just call `PyBytes_RESIZE`, which can do an in-place size reduction of the `bytes` object

Any excess memory goes back to the memory pool, which Python can use as it sees fit

Reply:
I have just tested this using the Mac 14 wheels from https://github.com/zarr-developers/numcodecs/actions/runs/14105743382 using the memray code in https://github.com/tomwhite/memray-array - and can confirm that it works. I can see the memory saving (peak memory goes form 300MB to 200MB). This was using zstd with Zarr v3.